server/graphql/types.js

```diff
@@ -66,12 +67,15 @@ const getCustomTypes = (strapi, nexus) => {
 							},
 						};
 
-						// only return published entries
-						if (_.get(contentType, ['options', 'draftAndPublish'], false)) {
-							query.publicationState = 'live';
+						// return entries based on publicationState arg or default to only returning published entries when draftAndPublish mode is enabled
+						if (_.get(contentType, ['options', 'draftAndPublish'], false) && publicationState) {
+							query.publicationState = publicationState;
+						} else if (_.get(contentType, ['options', 'draftAndPublish'], false)) {
+							query.publicationState = "live";
```

Comment:
I would make this a nested if instead of checking the same thing twice.

```suggestion
						// only return published entries by default if content type has draftAndPublish enabled
						if (_.get(contentType, ['options', 'draftAndPublish'], false)) {
							query.publicationState = 'live';
							if (publicationState) {
								query.publicationState = publicationState;
							}
						}
```